tests/tooltips/colorPicker_test.js

```diff
@@ -1,55 +1,78 @@
+        mockedColorPicker = getMockedTooltip(sandbox, tooltipClasses.colorPicker, ["detector", "initialize"])
+    });
+
+    after(function () {
+        sandbox.restore();
+        mockedColorPicker.remove();
+    });
 
     it("! Before open Paren", function() {
         var line = "fill(255, 244, 0, 21);";
         var pre = "fill";
-        expect(testMockedTooltipDetection(mockedColorPicker, line, pre)).to.be(false);
+        expect(testMockedTooltipDetection(sandbox, mockedColorPicker, line, pre)).to.be(false);
     });
 
     it("After open Paren", function() {
         var line = "fill(255, 244, 0, 21);";
         var pre = "fill(";
-        expect(testMockedTooltipDetection(mockedColorPicker, line, pre)).to.be(true);
+        expect(testMockedTooltipDetection(sandbox, mockedColorPicker, line, pre)).to.be(true);
     });
 
     it("Middle", function() {
         var line = "fill(255, 244, 0, 21);";
         var pre = "fill(255, ";
-        expect(testMockedTooltipDetection(mockedColorPicker, line, pre)).to.be(true);
+        expect(testMockedTooltipDetection(sandbox, mockedColorPicker, line, pre)).to.be(true);
     });
 
     it("Before close paren", function() {
         var line = "fill(255, 244, 0, 21);";
         var pre = "fill(255, 244, 0, 21";
-        expect(testMockedTooltipDetection(mockedColorPicker, line, pre)).to.be(true);
+        expect(testMockedTooltipDetection(sandbox, mockedColorPicker, line, pre)).to.be(true);
     });
 
     it("! After close paren", function() {
         var line = "fill(255, 244, 0, 21);";
         var pre = "fill(255, 244, 0, 21)";
-        expect(testMockedTooltipDetection(mockedColorPicker, line, pre)).to.be(false);
+        expect(testMockedTooltipDetection(sandbox, mockedColorPicker, line, pre)).to.be(false);
     });
 
     it("All function names", function() {
-        _.each(["fill", "background", "stroke", "color"], function(fn) {
+        ["fill", "background", "stroke", "color"].forEach(function(fn) {
             var line = fn + "();";
             var pre = fn + "(";
-            expect(testMockedTooltipDetection(mockedColorPicker, line, pre)).to.be(true);
+            expect(testMockedTooltipDetection(sandbox, mockedColorPicker, line, pre)).to.be(true);
         });
     });
 
     it("! Different function name", function() {
         var line = "rect(255, 244, 0, 21);";
         var pre = "rect(255, ";
-        expect(testMockedTooltipDetection(mockedColorPicker, line, pre)).to.be(false);
+        expect(testMockedTooltipDetection(sandbox, mockedColorPicker, line, pre)).to.be(false);
     });
 });
 
 
 
 describe("colorPicker - selection (what it replaces)", function() {
-    var mockedColorPicker = getMockedTooltip(tooltipClasses.colorPicker, ["detector", "updateText", "initialize"]);
+    let sandbox;
+    let colorPicker;
+
+    before(function () {
+        sandbox = sinon.sandbox.create();
+        colorPicker = getTooltip(tooltipClasses.colorPicker);
```

Comment:
This now uses getTooltip() which doesn't set up any internal fakes on the Tooltip class, it just constructs the relevant Tooltip. These tests don't require any internal faked methods.
(This change is also true in the other files, for these types of tests)

Reply:
Nice!  fwiw I like tests that use dependencies as is instead of trying to mock everything out.